catalog/secrets/api/src/main/java/org/projectnessie/catalog/secrets/BasicCredentials.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.secrets;
+
+import jakarta.annotation.Nonnull;
+import java.util.Map;
+
+/** Represents a name + secret pair. */
+public interface BasicCredentials extends Secret {
+  @Nonnull
+  String name();
```

Comment:
Generally speaking, would it be worth storing all fields of all secret types as `char[]` instead of `String`? This was done in the OAuth client, which also has a `Secret` class.

Reply:
I know, there's a pattern to use `char[]` - but I suspect it's rather not useful in practice, given that basically every usage needs a `String` anyways - so it's additional heap pressure (not an argument for security) but also secrets duplicated on the heap. Also, all secret manager APIs return secrets as strings - some even as JSON or `map<string,string>` :shrug: